src/main/java/net/minecraftforge/common/config/Configuration.java

```diff
@@ -1553,9 +1588,34 @@ public String getString(String name, String category, String defaultValue, Strin
      * @return The value of the new string property.
      */
     public String getString(String name, String category, String defaultValue, String comment, String[] validValues, String langKey)
+    {
+        return setPropertyAndGetString(name, category, defaultValue, comment, validValues, null, langKey);
+    }
+
+    /**
+     * Creates a string property.
+     *
+     * @param name Name of the property.
+     * @param category Category of the property.
+     * @param defaultValue Default value of the property.
+     * @param comment A brief description what the property does.
+     * @param validValues A list of valid values that this property can be set to.
+     * @param validValuesDisplay an array of the config GUI display versions of the valid values that this Property can be set to.
+     * @param langKey A language key used for localization of GUIs
+     * @return The value of the new string property.
+     */
+    public String getString(String name, String category, String defaultValue, String comment, String[] validValues, String[] validValuesDisplay, String langKey)
+    {
+        return setPropertyAndGetString(name, category, defaultValue, comment, validValues, validValuesDisplay, langKey);
+    }
+
+    private String setPropertyAndGetString(String name, String category, String defaultValue, String comment, String[] validValues, String[] validValuesDisplay, String langKey)
```

Comment:
```suggestion
    private String setPropertyAndGetString(String name, String category, String defaultValue, String comment, String[] validValues, @Nullable String[] validValuesDisplay, String langKey)
```
and add `import javax.annotation.Nullable;` at the top of the file

Reply:
I think it's worth first considering how to deal with the question I asked in the OP. As it relates here:
1. I only added the null check to validValuesDisplay because I made the constructors of `Property` initialize it to empty, rather than null. And I only did that because that's what's done in all the other constructors for other arrays, like validValues and defaultValues. But that flies in the face of the documentation for those arrays' getters, which says they will be null if not defined. So the null check prevents it from being changed from empty to null. In hind-site, however, it doesn't currently matter if it's null or empty, since checks for both cases are done in `GuiConfigEntries`'s constructor. Perhaps the null check should be removed.
2. The other arrays, like validValues and defaultValues, are already also nullable. In fact, classes like Property and Configuration are jam-packed with un-annotated nullable parameters. The only reason to add nullable exclusively to validValuesDisplay is the explicit null check, correct?

Reply:
For better or worse, the resolution that most closely mirrors what's already in the code base would be to simply:
1. Change the documentation of `Propery#getValidValues` and `Propery#getValidValuesDisplay` to indicate that the returned array will be null or empty if not defined.
2. Remove the null checks altogether (without annotating as nullable).

Although I'm fine with any other route. Your call.

Reply:
I think it's best to accurately describe the current behavior, so your solution is good.
I don't think it would be simple to fix it up to work more consistently without breaking things.

Reply:
Ok, that should do it.